### DISCUSSION TITLE
Adding Rental Products

### DISCUSSION THREAD

**Discussion by juanzgc** [saleor/saleor#8975] [Q&A]
Will Saleor enable products to also be structured as rental products? For example the e-commerce application that I need to create needs to allow for selling bike products, and rent road bikes.

As of now I know this isn't possible but will it possible every be implemented? Or is there an integration to enable this possibility?

I have seen a similar post: https://github.com/saleor/saleor/discussions/8912
But I do not plan on building a marketplace, so this doesn't quite answer my question in terms of enabling the product structure to enable rentals. So a specific product variant loses availability based on the date (One date it may be available, while another date isn't).

**Comment by JoshFialkoff** [comment]
Is there an update on this?

**Reply by korycins** [reply]
> Is there an update on this?

At this point there are no plans to build this feature inside Saleor.

> specific product variant loses availability based on the date

This should be possible by implementing a custom saleor app, that would provide the dates when the productVariant is available for renting. 
See more about [apps](https://docs.saleor.io/developer/extending/apps/overview)

**Comment by timuric** [comment]
Hey @juanzgc @JoshFialkoff , I believe what you are describing is outside the scope of a typical e-commerce platform. The process of renting is a sale, which you can model with Product attributes and within the business logic of your checkout. In terms of returns, you would have to create an app to track returns and extra charges. 

This is not generic enough to be an out-of-the-box feature. However, Saleor provides enough APIs and flexibility to combine your rental business logic and transactional functionality of the commerce platform
